### PR TITLE
feat: payments w tests and deploy script

### PR DESCRIPTION
**Pull request by 8sunyuan**
Reopening PR to view more cleanly. 
Draft PR here https://github.com/Layr-Labs/eigenlayer-contracts/pull/488
Includes unit tests and deployment scripts for Holesky.

Introduces `PaymentCoordinator` for AVSs to submit rangePayments for specified period of time and for operators/stakers to claim payments via merkle proofs.

### CODE REVIEW

File: src/contracts/core/PaymentCoordinator.sol [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+     * @notice Creates a new distribution root
+     * @param root The merkle root of the distribution
+     * @param paymentCalculationEndTimestamp The timestamp until which payments have been calculated
+     * @param activatedAt timestamp at which that the root can be claimed against
+     * @dev Only callable by the paymentUpdater
+     */
+    function submitRoot(
+        bytes32 root,
+        uint64 paymentCalculationEndTimestamp,
+        uint64 activatedAt
```
Comment (gpsanant) [inline src/contracts/core/PaymentCoordinator.sol:168]: remove

File: src/contracts/core/PaymentCoordinator.sol [hunk 2]
```diff
@@ -0,0 +1,490 @@
...
+        require(
+            rangePayment.duration <= MAX_PAYMENT_DURATION,
+            "PaymentCoordinator._payForRange: duration exceeds MAX_PAYMENT_DURATION"
+        );
+        require(
+            rangePayment.duration % calculationIntervalSeconds == 0,
+            "PaymentCoordinator._payForRange: duration must be a multiple of calculationIntervalSeconds"
+        );
+        require(
+            block.timestamp - MAX_RETROACTIVE_LENGTH <= rangePayment.startTimestamp &&
```
Comment (gpsanant) [inline src/contracts/core/PaymentCoordinator.sol:203]: need `require(rangePayment.startTimestamp % calculationIntervalSeconds == 0)`

Comment (8sunyuan) [reply src/contracts/core/PaymentCoordinator.sol:204]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/512/commits/b02c588081a4ce25e1f683513218acfb7e25f4ca#diff-2a28246efa54c09dff5f0afef55f490d130b1a13c5b7e30efb51d5647c52a1a2R200

File: src/contracts/core/PaymentCoordinator.sol (hunk 1 above)
Comment (8sunyuan) [reply src/contracts/core/PaymentCoordinator.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/512/commits/b02c588081a4ce25e1f683513218acfb7e25f4ca#diff-2a28246efa54c09dff5f0afef55f490d130b1a13c5b7e30efb51d5647c52a1a2L168

Removed the activatedAt

**Review by gpsanant — APPROVED** [review]

**Pull request merged**